docs/getting-started/configuration.md

```diff
@@ -0,0 +1,307 @@
+
+```shell
+# From your Backstage root directory
+cd packages/backend
+yarn add pg
+```
+
+Use your favorite editor to open `app-config.yaml` and add your PostgreSQL
+configuration. in the root directory of your Backstage app using the credentials
+from the previous steps.
+
+```diff
+backend:
+  database:
+-    client: sqlite3
+-    connection: ':memory:'
++    # config options: https://node-postgres.com/api/client
++    client: pg
++    connection:
++      host: ${POSTGRES_HOST}
++      port: ${POSTGRES_PORT}
++      user: ${POSTGRES_USER}
++      password: ${POSTGRES_PASSWORD}
++      # https://node-postgres.com/features/ssl
++      #ssl: require # see https://www.postgresql.org/docs/current/libpq-ssl.html Table 33.1. SSL Mode Descriptions (e.g. require)
++        #ca: # if you have a CA file and want to verify it you can uncomment this section
++        #$file: <file-path>/ca/server.crt
+```
+
+You'll use the connection details from the previous step. You can either set the
+`POSTGRES_` environment variables prior to launching Backstage, or remove the
+`${...}` values and set actual values in this configuration file.
+
+The default port for PostgreSQL is `5432` or `5433`, and the host name could be
+`127.0.0.1` if installed locally. A word of caution: In general, using
+connection details in a configuration file is not recommended.
+
+Start the Backstage app:
+
+```shell
+yarn dev
+```
+
+After Backstage is completely started you'll notice the catalog is populated
+with the information, still coming from the configuration files. If you add a
+new component, or register an existing one it will be saved in the database.
+Later in this tutorial you'll add a service, and you can test if it's persistent
+as advertised.
+
+If you want to read more about the database configuration, here's some helpful
+links:
+
+- [Configuring Plugin Databases](../tutorials/configuring-plugin-databases#privileges)
+- [Read more about Knex](http://knexjs.org/), which is the library we use for
+  the database backend
+
+### Setting up authentication
+
+There's multiple authentication providers available for you to use with
+Backstage, feel free to follow
+[the instructions for adding authentication](../auth/).
+
+For this tutorial we choose to use GitHub, a free service most of you might be
+familiar with. For other options, see
+[the auth provider documentation](../auth/github/provider#create-an-oauth-app-on-github).
+
+Go to
+[https://github.com/settings/applications/new](https://github.com/settings/applications/new)
+to create your OAuth App. The `Homepage URL` should point to Backstage's
+frontend, in our tutorial it would be `http://127.0.0.1:3000`. The
+`Authorization callback URL` will point to the auth backend, which will most
+likely be `http://127.0.0.1:7007/api/auth/github/handler/frame`.
+
+<p align='center'>
+  <img src='../assets/getting-started/gh-oauth.png' alt='Screenshot of the GitHub OAuth creation page'>
```

Comment:
```suggestion
  <img src='../assets/getting-started/gh-oauth.png' alt='Screenshot of the GitHub OAuth creation page' />
```

Docusarus 2 is complaining. Ref https://github.com/backstage/backstage/pull/8356